### PR TITLE
Add hpa test

### PR DESCRIPTION
**Pull request by totallyGreg**
This adds a test for making sure that a hpa name-<revision> has a spec.targetScaleRef.name that points to the same name -<revision> deployment.

This is the first step in trying to figure out where the discrepancy noticed in https://github.com/istio/istio/issues/28107 is coming from. 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by shamsher31 — APPROVED** [review]
Thanks.

File: operator/cmd/mesh/manifest-generate_test.go [hunk 1]
```diff
@@ -566,13 +566,15 @@ func TestConfigSelectors(t *testing.T) {
 		// First we fetch all the objects for our default install
 		cname := "istiod"
 		deployment := mustFindObject(t, objs, cname, name.DeploymentStr)
+		hpa := mustFindObject(t, objs, cname, name.HPAStr)
```
Comment (shamsher31) [inline operator/cmd/mesh/manifest-generate_test.go:569]: I think you can remove this line as it's not being used anywhere.

Comment (totallyGreg) [reply operator/cmd/mesh/manifest-generate_test.go:569]: I'm not sure I understand the overall testing strategy, as the "cross revision" run seems to want to set both a default and a canary and then verify things are not incorrect between them.  The hpa test is already being done in the validateRefererntialIntegrity function but only checking against cname.  I'll remove the extra hpa.

Comment (totallyGreg) [reply operator/cmd/mesh/manifest-generate_test.go]: /retest

**Comment by shamsher31** [comment]
@totallyGreg can you run `make gen` and `make refresh-goldens` and add any file changed?

**Comment by totallyGreg** [comment]
> @totallyGreg can you run `make gen` and `make refresh-goldens` and add any file changed?

I ran `make gen` and after a lot of fan noise it failed trying to execute the refresh-golden target itself. 
 
```makefile
FAIL
Makefile.core.mk:334: recipe for target 'refresh-goldens' failed
make[1]: *** [refresh-goldens] Error 1
make: *** [gen] Error 2
``` 
The only changed files are go.mod & go.sum.  
I can push these if needed.  

```shell
❯ git status
On branch add-hpa-test
Your branch is up to date with 'origin/add-hpa-test'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   go.mod
        modified:   go.sum
```

File: operator/cmd/mesh/manifest-generate_test.go [hunk 2]
```diff
@@ -573,6 +573,7 @@ func TestConfigSelectors(t *testing.T) {
 		// Next we fetch all the objects for a revision install
 		nameRev := "istiod-canary"
 		deploymentRev := mustFindObject(t, objsRev, nameRev, name.DeploymentStr)
+		hpaRev := mustFindObject(t, objs, nameRev, name.HPAStr)
```
Comment (shamsher31) [inline operator/cmd/mesh/manifest-generate_test.go:576]: @totallyGreg  After spending an hour trying to debug a test, found out that you were looking into the wrong object for the Rev label. I ran it locally and now  `make gen` and `make refresh-goldens` are passing.
```suggestion
		hpaRev := mustFindObject(t, objsRev, nameRev, name.HPAStr)
```

**Comment by shamsher31** [comment]
Not sure why the Docs team is asked to review this PR. @ericvn can you PTAL?

**Comment by totallyGreg** [comment]
> Not sure why the Docs team is asked to review this PR. @ericvn can you PTAL?

Probably because I was dumb and corrected a series of 3 unrelated typos in some comments.

**Comment by shamsher31** [comment]
@totallyGreg can you rebase your PR? I don't think `go.mod` and `go.sum` changes are needed in this PR.

**Comment by shamsher31** [comment]
cc: @istio/wg-docs-maintainers

**Review by ericvn — APPROVED** [review]

**Pull request merged**